buildSrc/src/main/java/org/springframework/boot/build/ConventionsPlugin.java

```diff
@@ -123,53 +116,81 @@ public void apply(Project project) {
 
 	private void applyJavaConventions(Project project) {
 		project.getPlugins().withType(JavaBasePlugin.class, (java) -> {
-			project.getPlugins().apply(TestFailuresPlugin.class);
+			applyTestFailuresPlugin(project);
 			configureSpringJavaFormat(project);
-			project.setProperty("sourceCompatibility", "1.8");
-			project.getTasks().withType(JavaCompile.class, (compile) -> {
-				compile.getOptions().setEncoding("UTF-8");
-				withOptionalBuildJavaHome(project, (javaHome) -> {
-					compile.getOptions().setFork(true);
-					compile.getOptions().getForkOptions().setJavaHome(new File(javaHome));
-					compile.getOptions().getForkOptions().setExecutable(javaHome + "/bin/javac");
+			setSourceCompatibility(project);
```

Comment:
I'd prefer to set the property directly here rather than wrapping it in a method call.